src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -877,5 +887,75 @@ public async Task<List<GitFile>> GetCommonScriptFilesAsync(string repoUri, strin
+
+            JObject globalJson = await _fileManager.ReadGlobalJsonAsync(repoUri, commit);
+            JToken dotnet = globalJson.SelectToken("tools.dotnet", true);
+
+            _logger.LogInformation("Reading dotnet version from global.json succeeded!");
+
+            SemanticVersion.TryParse(dotnet.ToString(), out SemanticVersion dotnetVersion);
+
+            if (dotnetVersion == null)
+            {
+                _logger.LogError($"Failed to parse dotnet version from global.json from repo: {repoUri} at commit {commit}. Version: {dotnet.ToString()}");
+            }
+
+            return dotnetVersion;
+        }
+
+        /// <summary>
+        /// Updates the global.json entries for tools.dotnet and sdk.version if they are older than an incoming version
+        /// </summary>
+        /// <param name="incomingDotnetVersion">version to compare against</param>
+        /// <param name="repoGlobalJson">Global.Json file to update</param>
+        /// <returns>Updated global.json file if was able to update, or the unchanged global.json if unable to</returns>
+        private GitFile UpdateDotnetVersionGlobalJson(SemanticVersion incomingDotnetVersion, GitFile repoGlobalJson)
+        {
+            string repoGlobalJsonContent = repoGlobalJson.ContentEncoding == ContentEncoding.Base64 ?
+                _gitClient.GetDecodedContent(repoGlobalJson.Content) :
+                repoGlobalJson.Content;
+            try
+            {
+                JObject parsedGlobalJson = JObject.Parse(repoGlobalJsonContent);
+                if (SemanticVersion.TryParse(parsedGlobalJson.SelectToken("tools.dotnet").ToString(), out SemanticVersion repoDotnetVersion))
+                {
+                    if (repoDotnetVersion.CompareTo(incomingDotnetVersion) < 0)
+                    {
+                        parsedGlobalJson["tools"]["dotnet"] = incomingDotnetVersion.ToNormalizedString();
+
+                        // Also update and keep sdk.version in sync.
+                        JToken sdkVersion = parsedGlobalJson.SelectToken("sdk.version");
+                        if (sdkVersion != null)
+                        {
+                            parsedGlobalJson["sdk"]["version"] = incomingDotnetVersion.ToNormalizedString();
+                        }
+                        return new GitFile(VersionFiles.GlobalJson, parsedGlobalJson);
+                    }
+                    return repoGlobalJson;
+                }
+                else
+                {
+                    _logger.LogInformation("Could not parse the repo's dotnet version from the global.json. Skipping update to dotnet version sections");
+                    return repoGlobalJson;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation($"Failed to update Dotnet version for global.json. Skipping update to version sections. Exception: {ex}");
```

Comment:
SHould this be an error? (and the exception can be logged with Log...(ex, blah)

Reply:
Yeah, I guess I didn't want it to show up as an error since both the repo or arcade's global.json could be broken by a bad checkin, but it's worth having the error to look at when investigating. I'll change it to log an error.